lychee-bin/src/commands/check.rs

```diff
@@ -135,10 +135,14 @@ async fn handle(client: &Client, cache: Arc<Cache>, request: Request) -> Respons
     // TODO: Handle error as soon as https://github.com/seanmonstar/reqwest/pull/1399 got merged
     let response = client.check(request).await.expect("cannot check URI");
 
-    // Never cache filesystem access as it is fast already so caching has no
-    // benefit
-    if !uri.is_file() {
-        cache.insert(uri, response.status().into());
+    // - Never cache filesystem access as it is fast already so caching has no
+    //   benefit.
+    // - Skip caching unsupported URLs as they might be supported in a
+    //   future run.
+    // - Skip caching excluded links; they might not be excluded in the next run
+    let status = response.status();
+    if !uri.is_file() && !status.is_excluded() && !status.is_unsupported() {
```

Comment:
Could similar logic be used to disable caching when the `Status` has been overridden through the `--accept` flag? Or is that the wrong way to solve the conflict between `--accept` and `--cache` raised here:  https://github.com/lycheeverse/lychee/pull/685#discussion_r919501382 ?

Reply:
TBH I don't know what's the use case of --accept..

Reply:
It was mostly a workaround for the dreaded 429 status code because we don't support rate limiting yet.
That said, some websites send special status codes (e.g. LinkedIn uses 999), that we don't know how to interprete. In such cases it's best to let the user decide by overwriting the default accepted status codes.

Reply:
@walterbm, I had another look. Our caching takes the accepted status codes into account. Consider this constructor for the status code:
https://github.com/lycheeverse/lychee/blob/a557cba0b4fcf63c496b9dc202787e899d290e08/lychee-lib/src/types/status.rs#L72-L85
The status only gets overridden if `accepted.map(|a| a.contains(&code))`. The remaining status codes remain untouched. So for example if a user sets `--accept 123,456`, then these status codes will be set to `Status::Ok(code)` on top of the normally accepted ones.

To answer your question, yes I think we could add the check for `--accept` here and maybe it's fine if we just skip caching the user-specified status codes. These could change from run to run.

Reply:
@mre ah that makes a lot of sense! Thank you for this context!